typed-racket-lib/typed/untyped-utils.rkt

```diff
@@ -39,19 +39,18 @@
                     [typed-module  (generate-temporary #'typed-module)]
                     [untyped-module  (generate-temporary #'untyped-module)]
                     [*racket/base (datum->syntax #'from-module-spec 'racket/base)]
-                    [*typed/racket/base (datum->syntax #'from-module-spec
-                                                       'typed/racket/base)]
-                    [*require (datum->syntax #'from-module-spec
-                                             'require)]
+                    [*typed/racket (datum->syntax #'from-module-spec (syntax-e #'language-spec))]
+                    [*require (datum->syntax #'from-module-spec 'require)]
+                    [*language-spec (datum->syntax #'racket/base (syntax-e #'language-spec))]
```

Comment:
This was `typed/racket` before. I think it needs to be the language-spec.

Reply:
@bennn Thanks for the feedback. I've changed the `datum->syntax` context to `#'_`. I'm not really up to good coding style conventions for Racket, so I hope I've done the right thing.

As for the error about an `(only-in quote)`, I agree the message could be improved, but from a cursory glance at the source, it would seem to require patching racket/collects/racket/private/reqprov.rkt (outside the `typed-racket` package). I'm also quite unsure of how to approach that. Somehow make `require/untyped-contract` a special case ?

Is there anything else I can do to improve this PR ?

Reply:
how about changing the syntax from `typed/racket/shallow` to `#:lang typed/racket/shallow` --- from `:id` to keyword +`:expr` ? that should improve the error and make this easier to use later.